build_modules/lib/src/module_cache.dart

```diff
@@ -63,17 +64,18 @@ class DecodingCache<T> {
     if (!_cached.containsKey(id)) {
       entry = _cached[id] = _Entry()
         ..needsCheck = false
-        ..value = Result.capture(reader.readAsBytes(id).then(_fromBytes))
-        ..digest = Result.capture(reader.digest(id));
+        ..value =
+            Result.capture(toFuture(reader.readAsBytes(id)).then(_fromBytes))
```

Comment:
I wonder if it would be sensible for `Result.capture` to take a `FutureOr` even outside of this specific use.

Reply:
that might be nice yes